src/Data/Languages/Templates.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DeriveLift #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Data.Languages.Templates
+  ( Language (..),
+    LanguageKey,
+    generateLanguageMap,
+  )
+where
+
+import Data.Aeson
+import Data.ByteString (ByteString)
+import qualified Data.Map.Strict as Map
+import Data.Text (Text)
+import Data.Yaml
+import GHC.Generics (Generic)
+import Language.Haskell.TH
+import Language.Haskell.TH.Syntax
+#ifndef LANGUAGES_YAML_PATH
+import Paths_lingo
+#endif
+
+-- | Type synonym for linguist's language name key.
+type LanguageKey = Text
+
+-- | Identifies a programming language.
+data Language = Language
+  { languageId :: Integer,
+    languageName :: LanguageKey,
+    languageExtensions :: [Text],
+    languageFileNames :: [Text]
+  }
+  deriving (Eq, Show, Lift)
+
+instance FromJSON Language where
+  parseJSON = withObject "Language" $ \l ->
+    Language
+      <$> l .: "language_id"
+      <*> pure "<ERROR: Unnamed language (this is a bug in Template Haskell)"
+      <*> l .:? "extensions" .!= []
+      <*> l .:? "filenames" .!= []
```

Comment:
I don’t think this instance needs to exist. Notably, the instance in the previous code was never part of the published code, only the generating code. Further, this instance looks to be incorrect since it will always fail.

Reply:
It is used in the YAML deserialization; `libyaml` piggybacks upon aeson. I do, however, agree that the `error` call is fixable.